src/index-format/dune

```diff
@@ -0,0 +1,10 @@
+(library
+ (name merlin_index_format)
```

Comment:
Why the name `index_format`?

Reply:
This was answered offline: it uses the same convention as compiler artifacts: `cmt_format`, `cmi_format` etc.